### PR TITLE
planner: choose non-prefix column when both index key and handle have the same one

### LINKED ISSUE #22960: prefix column in secondary index in clustered index table maybe return wrong result
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
set @@tidb_enable_clustered_index=1;
create table t(c1 varchar(100), c2 varchar(100), c3 varchar(100), primary key (c1,c2), key idx1 (c2(1)));
insert into t select 'a', 'cd', 'ef';
select c2 from t use index(idx1);
select count(1) from t use index(idx1) where c2 = 'cd';
```

### 2. What did you expect to see? (Required)

```
cd
1
```

### 3. What did you see instead (Required)

```
c
0
```
### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by lysu**
Signed-off-by: lysu <sulifx@gmail.com>

### What problem does this PR solve?

Issue Number: close #22960

Problem Summary:

user can create a index contain two same column values in one index with clustered index support.

for example, we have a table `t`

```
create table t(
  c1 varchar(100), 
  c2 varchar(100), 
  c3 varchar(100), 
  primary key (c1,c2) clustered, 
  key idx1 (c2(1)),
)
```

then for an IndexReader Query:

```
select c2 from t use index(idx1) where c = 'x';
```

IndexQuery will have information about 2 `c2`, one from index value, and another from primary key value(CommonHandle)

if there are no prefix column, operator above IndexReader can choose any of those two columns

but if there is a prefix column like this table schema, planner need make all operators above is use non-prefix column

### What is changed and how it works?

What's Changed:

make ColumnIndex to prefer choose full column

### Related changes

- n/a

### Check List 

Tests 

- Unit test
- 
Side effects

- n/a

### Release note 

- n/a

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/22980)

### CODE REVIEW

**Comment by lysu** [comment]
/cc @winoros @lzmhhh123  PTAL

I'm not sure whether it's a right way to modify such low-level method, help to take look, thanks

File: executor/clustered_index_test.go [hunk 1]
```diff
@@ -51,6 +51,41 @@ func (s *testClusteredSuite) TestClusteredUnionScan(c *C) {
 	tk.MustExec("rollback")
 }
 
+func (s *testClusteredSuite) TestClusteredPrefixColumn2(c *C) {
+	tk := s.newTK(c)
+	tk.MustExec("drop table if exists t1, t2")
+
+	tk.MustExec("create table t1(c1 varchar(100), c2 varchar(100), c3 varchar(100), primary key (c1,c2), key idx1 (c2(1)))")
+	tk.MustExec("insert into t1 select 'a', 'cd', 'ef'")
+	tk.MustQuery("explain select c2 from t1 use index(idx1)").Check(testkit.Rows(
```
Comment (lzmhhh123) [inline executor/clustered_index_test.go:60]: Please move the plan tests to `testdata` file.

File: expression/column.go [hunk 2]
```diff
@@ -200,6 +200,7 @@ type Column struct {
 
 	OrigName string
 	IsHidden bool
+	IsPrefix bool
```
Comment (lzmhhh123) [inline expression/column.go:203]: We'd better add a comment for the clustered index case, such as:
```
pk(col1, col2), index(col1(10)), key: col1(10)_col1_col2
pk(col1(10), col2), index(col1), key: col1_col1(10)_col2
``
to explain it.

**Review by lzmhhh123 — COMMENTED** [review]
Rest LGTM

**Review by lzmhhh123 — APPROVED** [review]

**Comment by coocood** [comment]
LGTM

**Pull request merged**